src/feature/control/control_hs.c

```diff
@@ -134,6 +134,13 @@ handle_control_onion_client_auth_add(control_connection_t *conn,
         }
       } SMARTLIST_FOREACH_END(flag);
     }
+    if (!strcasecmp(line->key, "ClientName")) {
+      if (strlen(line->value) > REND_CLIENTNAME_MAX_LEN) {
+        control_printf_endreply(conn, 512, "ClientName longer than %d chars",
+                                REND_CLIENTNAME_MAX_LEN);
+      }
+      strlcpy(creds->client_name, line->value, strlen(line->value) + 1);
```

Comment:
Hmm, I don't like this `strlen() + 1`.
Let's instead change it to use the `sizeof()` of the destination like it happens a bit above:
`  strlcpy(creds->onion_address, hsaddress, sizeof(creds->onion_address));`